### ISSUE TITLE
Missing implementation of integrator init for PDEProblem

### ISSUE THREAD

**Issue by bgroenks96** [SciML/DiffEqBase.jl#628]
Trying to use the integrator interface via `init` on a `PDEProblem` results in the following error:

```
type PDEProblem has no field p
in top-level scope at heat_conduction.jl:59
in  at DiffEqBase/znIav/src/solve.jl:37
in #init#446 at DiffEqBase/znIav/src/solve.jl:43
in  at DiffEqBase/znIav/src/solve.jl:135
in #get_concrete_problem#457 at DiffEqBase/znIav/src/solve.jl:135
in get_concrete_p at DiffEqBase/znIav/src/solve.jl:208
in getproperty at base/Base.jl:33
```

This can be corrected by adding the following function implementation to `solve.jl`, mirroring that of `solve`:

```julia
function DiffEqBase.init(prob::PDEProblem,alg::DiffEqBase.DEAlgorithm,args...;
                                          kwargs...)
    init(prob.prob,alg,args...;kwargs...)
end
```

Pull request to follow shortly.

**Comment by ChrisRackauckas** [comment]
There is no integrator for a PDEProblem. This won't work. It needs to be `discretize`d

**Comment by bgroenks96** [comment]
I don't know what you mean. The result of `discretize` is a `PDEProblem` with a nested `ODEProblem`. The implementation of `init` should, as far as I can tell, look the same as the one for `solve`.

I tested this on a `PDEProblem` obtained and from `discretize` and it works exactly as expected.

**Comment by devmotion** [comment]
In any case, the `DiffEqBase` qualifications could be removed both in `solve` and a potential implementation of `init`. I assume the code was just copied from an external package at some point.

**Comment by bgroenks96** [comment]
@devmotion I was wondering why that was there!

**Comment by bgroenks96** [comment]
Fixed by #636
